L106-node-heath-check-library.md

```diff
@@ -0,0 +1,70 @@
+Node Heath Check Library 2.0
+----
+* Author(s): murgatroid99
+* Approver: wenbozhu
+* Status: In Review
+* Implemented in: Node.js
+* Last updated: 2023-08-25
+* Discussion at: https://groups.google.com/g/grpc-io/c/6y632NLeUTw
+
+## Abstract
+
+This document proposes a new, implementation-agnostic API for the Node [grpc-health-check](https://www.npmjs.com/package/grpc-health-check) package.
+
+## Background
+
+The current grpc-health-check package is old (the latest release was in 2020), and its API and design negatively impacts usability in a few ways:
+ - Most importantly, it directly depends on the old [grpc](https://www.npmjs.com/package/grpc) package, and inherets all of the platform compatibility problems that contributed to that package's deprecation in 2021.
+ - The package has no TypeScript type information, which has become much more important in recent years.
+ - The basic functionality of indicating the serving status of a service involves referring to a protobuf enum deep in a package hierarchy (e.g. `grpcHealthCheck.messages.HealthCheckResponse.ServingStatus.SERVING`)
+ - The package exports a `Client` object that requires a specific gRPC implementation to be determined in advance, and it uses protoc-generated types, which are less commonly used and can be awkward to work with.
+
+### Related Proposals:
+* [gRFC A17: Client-Side Health Checking](https://github.com/grpc/proposal/blob/master/A17-client-side-health-checking.md)
+
+## Proposal
+
+We will publish version 2.0 of the grpc-health-check library with the following API, defined in TypeScript:
+
+```ts
+type ServingStatus = 'UNKNOWN' | 'SERVING' | 'NOT_SERVING';
+interface ServingStatusMap {
+  [serviceName: string]: ServingStatus;
+}
+
+class HealthImplementation {
```

Comment:
Do we need to call out that this is server-only, e.g. ServerHealthImplementation ?

Reply:
The current API uses a similar name to refer to an object with similar functionality, and no user that I have seen has expressed any confusion, so I think the name is fine as it is.